src/mlpack/methods/ann/layer/batch_norm_impl.hpp

```diff
@@ -216,26 +216,26 @@ void BatchNormType<MatType>::Forward(
     // Calculate mean and variance over all channels.
     MatType mean = arma::sum(arma::sum(inputTemp, 2), 0) / m;
     variance = arma::sum(arma::sum(arma::pow(
-        inputTemp.each_slice() - arma::repmat(mean,
+        inputTemp.each_slice() - arma::repmat<MatType>(mean,
```

Comment:
Not sure why `arma::repmat()` would need to explicitly have the `<MatType>` template parameter specified.  The function is designed to automatically work with all Armadillo matrix types.

Reply:
I know, I was getting a weird error at some point and I had to fix two other things, I can removed it and see if this work back.

Reply:
Done @conradsnicta

Reply:
We might need it later to forward it to the correct backend (armadillo / bandicoot). But looking at the code, we have to make quite a few changes to make it work with both backends, so probably not needed to do it now.